File: KnowledgeSharingApi/KnowledgeSharingApi.Domain/Models/Entities/Entities/CourseRegister.cs
using KnowledgeSharingApi.Domains.Enums;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace KnowledgeSharingApi.Domains.Models.Entities.Entities
{
    public class CourseRegister : Entity
    {
        [Key]
        public Guid CourseRegisterId { get; set; }
        public Guid UserId { get; set; }
        public Guid CourseId { get; set; }


        protected override CourseRegister Init()
        {
            return new CourseRegister();
        }
    }
}

File: KnowledgeSharingVue/knowledge-sharing-vue/src/js/models/entities/course-register.js


import Entity from './entity';

class CourseRegister extends Entity {
    constructor() {
        super();
        this.CourseRegisterId = null;
        this.UserId = null;
        this.CourseId = null;
    }

    init() {
        return new CourseRegister();
    }
}

module.exports = CourseRegister;
